security/seccomp-profiles.adoc

```diff
@@ -0,0 +1,77 @@
+
+toc::[]
+
+A container or an OpenShift pod runs a single application that performs one or more well-defined tasks. The application usually requires only a small subset of the underlying operating system kernel APIs.
+Seccomp (secure computing mode) Linux kernel feature can be used to limit the process running in a container to only call a subset of the available system calls. These system calls can be configured by creating profile that is named as seccomp profile.
+
+?? Are below statements important or just a note?
+[IMPORTANT]
+====
+CRI-O runs unconfined by default, without any seccomp profile applied.
+====
+
+[IMPORTANT]
+====
+Seccomp profile cannot be applied to privileged containers.
+====
+
+== Configuring default seccomp profile
+Openshift ships with a default seccomp profile. On RHEL CoreOS, the default seccomp profile is stored in `/etc/crio/seccomp.json`. You can enable the default seccomp profile by overriding the empty seccomp value with `runtime/default`.
+
+
+.Sample default seccomp profile
+[%collapsible]
+====
+[source,json]
+----
+include::https://raw.githubusercontent.com/containers/common/main/pkg/seccomp/seccomp.json[]
+----
+//Add link to the appendix above
+//Alternate link found https://github.com/moby/moby/blob/master/profiles/seccomp/default.json
+====
+== Configuring a custom seccomp profile
+You can configure a custom seccomp profile, which allows you to update the filters based on the application requirements. This allows cluster administrators to have greater control over the security of workloads running in OpenShift Container Platform.
+[NOTE]
+====
+You can refer to default seccomp profile stored in `/etc/crio/seccomp.json` to create a custom seccomp profile.
+====
+
+.Prerequisite
+* You have created a custom SCC. For more information, see "Additional resources".
+* You have created a custom seccomp profile.
+
+.Procedure
+. Upload your custom seccomp profile to `/var/lib/kubelet/seccomp-<custom-name>.json` by using the Machine Config. See link:https://github.com/openshift/machine-config-operator#applying-configuration-changes-to-the-cluster[Machine Config Operator (MCO) documentation] for detailed steps.
+
+. Update the custom SCC by providing reference to the created custom seccomp profile:
+
+[source, json]
+----
+seccompProfiles:
+- localhost/seccomp-<custom-name>.json
+----
+
+During deployment, the admission controller performs the following checks:
+
+* Validates the annotations against the current SCCs allowed by the user role.
+* Validates the SCC, which includes the seccomp profile, is allowed for the pod.
+If the SCC is allowed for the pod, the kubelet runs the pod with the specified seccomp profile.
+
+[IMPORTANT]
+====
+Ensure that the seccomp profile is deployed to all worker nodes.
+====
+
+[NOTE]
+====
+The custom SCC must have the appropriate priority to be automatically assigned to the pod or meet other conditions required by the pod, such as allowing CAP_NET_ADMIN.
+====
+
```

Comment:
I think we should provide an example how to apply the seccomp profile to the workload, right?

Reply:
@saschagrunert Yes, an example will definitely help. Do you have a snippet that I can use?

Reply:
Yeah, how about a modified version of: https://kubernetes.io/docs/tutorials/clusters/seccomp/#create-a-pod-with-a-seccomp-profile-for-syscall-auditing

We should add both cases, the annotations (OpenShift < 4.6) and the preferred field (later OpenShift versions)

Reply:
Docs team does not support versions < 4.6, so I think I will not be touching upon the annotations. 
How about this https://kubernetes.io/docs/tutorials/clusters/seccomp/#create-pod-with-seccomp-profile-that-only-allows-necessary-syscalls for custom secomp profile 

and

https://kubernetes.io/docs/tutorials/clusters/seccomp/#create-pod-that-uses-the-container-runtime-default-seccomp-profile for RuntimeDefault?

Reply:
Yes, sounds good to me :)

Reply:
The field mentioned in json provided in the shared Google doc is represented differently in kubernetes doc. May I know if the json should look as-is  or are there any modifications necessary?

Reply:
This is available in k8s link: 
    seccompProfile:
      type: Localhost
      localhostProfile: profiles/fine-grained.json --> will rename this to custom-name

in the doc it shows
seccompProfiles:
- localhost/seccomp-<custom-name>.json

This is in reference to editing the custom SCC.

Reply:
ok. So when you say 'changing pod manifest' I believe it is same as annotation. So replacing the current annotation example with the detailed manifest from the k8s works fine?

Reply:
Changing the pod manifest means either using the annotation (deprecated) or the securityContext field.

Reply:
In custom seccomp profile procedure [here](https://deploy-preview-35859--osdocs.netlify.app/openshift-enterprise/latest/security/seccomp-profiles.html#configuring-a-custom-seccomp-profile),  can you confirm this: steps 1 & 2 are performed by developer and step 3 is performed by admin?

Reply:
The other way around: 1 & 2 are done by the system admin and 3 by the developer of the application.